plugins/autowarmth.koplugin/main.lua

```diff
@@ -0,0 +1,780 @@
+            else
+                menu[i].mode = nil
+            end
+        end
+    end
+    return menu
+end
+
+local about_text = _([[Set the frontlight warmth and night mode based on a time schedule or the sun's position.
+
+There are three types of twilight:
+• Civil: You can read a newspaper.
+• Nautical: You can see the first stars.
+• Astronomical: It is really dark.
+
+Certain warmth-values can be set for every kind of twilight and sunrise, noon, sunset and midnight.
+The screen warmth is continuously adjusted to the current time.
+
+To use the sun's position, the geographical location must be entered. The calculations are very precise (deviation less than two minutes).]])
+function AutoWarmth:getSubMenuItems()
+    return {
+        {
+            text = _("About auto-warmth"),
+            callback = function()
+                UIManager:show(InfoMessage:new{
+                    text = about_text,
+                     width = math.floor(Screen:getWidth() * 0.9),
+                })
+            end,
+            keep_menu_open = true,
+            separator = true,
+        },
+        {
+            text = _("Easy mode"),
+            checked_func = function()
+                return self.easy_mode
+            end,
+            help_text = _("Easy mode disables all entries except: civil dawn, sun rise, sun set and civil dusk."),
+            callback = function(touchmenu_instance)
+                local info_text
+                if not self.easy_mode then
+                    info_text = _("Reduce entries in auto-warmth schedule?")
+                else
+                    info_text = _("Expand entries in auto-warmth schedule?")
+                end
+                UIManager:show(ConfirmBox:new{
+                    text = info_text,
+                    ok_callback = function()
+                        self.easy_mode = not self.easy_mode
+                        G_reader_settings:saveSetting("autowarmth_easy_mode", self.easy_mode)
+                        self:scheduleMidnightUpdate()
+                        -- closing menu is necessary for refreshing the menu structure
+                        if touchmenu_instance then touchmenu_instance:closeMenu() end
+                    end,
+                })
+            end,
+            keep_menu_open = true,
+        },
+        {
+            text = _("Activate"),
+            checked_func = function()
+                return self.activate ~= 0
+            end,
+            sub_item_table = self:getActivateMenu(),
+        },
+        {
+            text = _("Location"),
+            enabled_func = function() return self.activate ~= activate_schedule end,
+            sub_item_table = self:getLocationMenu(),
+        },
+        {
+            text = _("Schedule"),
+            enabled_func = function() return self.activate ~= activate_sun end,
+            sub_item_table = self:getScheduleMenu(),
+        },
+        {
+            text = _("Warmths"),
```

Comment:
Warmth settings might be clearer?

Reply:
Should we also write `Scheduler settings` and `Location settings`?

Reply:
It's more that warmths feels odd, possibly even incorrect.